pkg/ipmasq/cell/cell.go

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import (
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/cilium/pkg/ipmasq"
+	ipmasqmaps "github.com/cilium/cilium/pkg/maps/ipmasq"
+	"github.com/cilium/cilium/pkg/option"
+
+	"github.com/cilium/hive/cell"
+)
+
+var Cell = cell.Module(
+	"ip-masq-agent",
+	"BPF ip-masq-agent implementation",
+
+	cell.Provide(newIPMasqAgentCell),
+	cell.Invoke(registerIPMasqAgentLifecycle),
+	cell.Config(defaultConfig),
+)
+
+type ipMasqAgentParams struct {
+	cell.In
+
+	Logger    *slog.Logger
+	Lifecycle cell.Lifecycle
+	Config    Config
+	IPMasqMap *ipmasqmaps.IPMasqBPFMap
+}
+
+type ipMasqAgentOut struct {
+	cell.Out
+
+	IPMasqAgent *ipmasq.IPMasqAgent
+}
+
+func newIPMasqAgentCell(params ipMasqAgentParams) (ipMasqAgentOut, error) {
+	if !option.Config.EnableIPMasqAgent {
+		return ipMasqAgentOut{}, nil
+	}
```

Comment:
If you move the `option.EnableIPMasqAgent` to the Cell, you can check the `params.Config.EnableIPMasqAgent` here. 

(For the future reference, even if you need to use DaemonConfig, you shouldn't refer the global variable in the option package here. Instead, you can add *option.DaemonConfig into the parameter and let Hive to inject it)